File: Algorithms/Math/GCD(x, y).cpp
#include <iostream>;

using namespace std;

int GCD(int x, int y);
int input_and_check(string print_text);

int main() {
	int x, y;
	cout << "Please enter two integers to get their GCD" << endl;
	x = input_and_check("num1 = ");
	y = input_and_check("num2 = ");
	cout << "GCD(" << x << ", " << y << ") = " << GCD(x, y);
	return 0;
}

int input_and_check(string print_text) {
	int x = -1;
	cout << print_text;
	cin >> x;
	while (!cin.good() || x < 2) {
		cout << "The input you entered is incorrect!" << endl << "Please try again" << endl;
		cin.clear();
		cin.ignore(INT_MAX, '\n');
		cin >> x;
	}
	return x;
}

int GCD(int x, int y) {
	int m, n, old_n;

	if (x > y) {m = x; n = y;} else {m = y; n = x;}

	while (n > 0) {
		old_n = n;
		n = m % n;
		m = old_n;
	}
	return m;
}

File: Algorithms/Math/GCD(x, y).py
import math


class GCD(object):

    @staticmethod
    def gcd(x, y):  # if GCD(x, y) = 1 then x and y  are relatively prime integers!
        if x < 0 or y < 0:
            print("Do not enter negative numbers!")
            return None
        else:
            if x > y:
                m = x
                n = y
            else:
                m = y
                n = x
            while n > 0:
                old_n = n
                n = m % n
                m = old_n
            return m


def main():  # test
    print("Please enter two integers to get their GCD")
    num1 = int(input("num1 = "))
    num2 = int(input("num2 = "))
    print(f"GCD({num1}, {num2}) = {GCD.gcd(num1, num2)}")
    return 0


if __name__ == '__main__':
    main()
